pkg/reconciler/ksvc/resources/knative_configuration.go

```diff
@@ -0,0 +1,80 @@
+package resources
+
+import (
+	"fmt"
+	"github.com/knative/serving/pkg/apis/autoscaling"
+	knservingv1alpha1 "github.com/knative/serving/pkg/apis/serving/v1alpha1"
+	"github.com/kubeflow/kfserving/pkg/apis/serving/v1alpha1"
+	"github.com/kubeflow/kfserving/pkg/constants"
+	"github.com/kubeflow/kfserving/pkg/frameworks/tensorflow"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func CreateModelServingContainer(modelName string, modelSpec *v1alpha1.ModelSpec) *v1.Container {
+	if modelSpec.Tensorflow != nil {
+		return tensorflow.CreateTensorflowContainer(modelName, modelSpec.Tensorflow)
+	} else {
+		//TODO(@yuzisun) handle other model types
+		return &v1.Container{}
+	}
+}
+
+func CreateKnativeConfiguration(kfsvc *v1alpha1.KFService) (*knservingv1alpha1.Configuration, *knservingv1alpha1.Configuration) {
+	var canaryContainer *v1.Container
+	defaultContainer := CreateModelServingContainer(kfsvc.Name, &kfsvc.Spec.Default)
+	if kfsvc.Spec.Canary != nil {
+		canaryContainer = CreateModelServingContainer(kfsvc.Name, &kfsvc.Spec.Canary.ModelSpec)
+	}
+	annotations := make(map[string]string)
```

Comment:
If you separate canary from default, you can just set annotations in place right, or is there some blocker?

Reply:
oh here we need to merge the annotations from kfservice and knative min/max scaler annotations to the configuration.

Reply:
I guess I'm confused why that can't happen in place. metadata.annotations is already initialized as a map, right? Should just be able to key them in, which might be a tad cleaner. Maybe I'm missing something. Either way, not a big deal at all, I trust your judgement, I just wanted to bring it up since it seemed like a little unnecessary complexity.

Reply:
metadata.annotations is nil, that's actually the reason why I did this way :)